roles/libvirt_manager/tasks/start_manage_vms.yml

```diff
@@ -23,19 +23,24 @@
     index_var: vm_id
     label: "{{ vm_type }}-{{ vm_id }}"
 
+- name: "Get public network XML"
+  register: public_net_xml
+  community.libvirt.virt_net:
+    command: get_xml
+    name: "{{ public_net_nics[0].network }}"
+
+- name: "Convert public network XML to dict"
+  ansible.builtin.set_fact:
+    public_net_dict: "{{ public_net_xml.get_xml | ansible.utils.from_xml }}"
```

Comment:
we can push this as a `vars` in the task below, since this seems to be the only place where we use it.
Also, we should be able to replace the `shell` and the `|jq` with proper ansible, but we can refine later (ansible-lint will probably crash due to the unsafe pipe).

Reply:
Good idea on moving that public_net_dict to a variable.

I also found a bunch other places that needed updating.
Not sure how to replace that jq select with something nice in ansible. Can we do follow up? (I think the no-qa thing will take care of ansible-lint for now?)

Reply:
sure, a follow-up is possible - add a todo just to be sure we don't forget about it.

IIRC you can just add `set -o pipefail;` for the ansible-lint to be happy.